addons/mail/static/src/core/web/messaging_menu.js

```diff
@@ -238,6 +240,18 @@ export class MessagingMenu extends Component {
         }
         return value;
     }
+
+    get displayChannelSelector() {
+        return this.state.addingChannel || this.state.addingChat;
+    }
+
+    get displayStartMeeting() {
+        return !this.state.addingChat && !this.state.addingChannel;
+    }
+
+    get displayStartConversation() {
+        return this.store.discuss.activeTab !== "channel" && !this.state.addingChat;
+    }
```

Comment:
This is quite painful to make 3 getters that need to be overridden just to add a new search.

This would be much simpler if there was just a `state.adding` that, when truthy means there's on ongoing `adding` process. The actual value is an id to the type of adding.

If you implement it this way, then whatsapp only has to add its `state.adding` with id `whatsapp`. The displaying of start a meeting or channel selector or start conversation, in terms of initial definition, remains unchanged as it should.

Reply:
I managed to get rid of the two getters, but we can't eliminate the third one because we can't replace the t-if condition for that button with `activeTab === 'chat' or activeTab === 'livechat' or activeTab === 'main'` or `activeTab !=='channel' and activeTab !== 'whatsapp'`